presentation/frontend/src/presentation-frontend/PresentationManager.ts

```diff
@@ -415,7 +406,7 @@ export class PresentationManager implements IDisposable {
   public async getNodesDescriptor(
     requestOptions: HierarchyLevelDescriptorRequestOptions<IModelConnection, NodeKey, RulesetVariable> & ClientDiagnosticsAttribute,
   ): Promise<Descriptor | undefined> {
-    await this.onConnection(requestOptions.imodel);
+    await this.ensureiModelInitialized(requestOptions.imodel);
```

Comment:
what do you think about this?

```ts
this.startiModelInitialization(requestOptions.imodel);
try {
  // whatever
} finally {
  await this.ensureiModelInitialized(requestOptions.imodel);
}
```

I think this would allow us to do our own async work while imodel initialization is happening at the same time, and in the `finally` block we would have to wait less compared to doing that before our request.